nisapi/clean/k4cb_dxd7.py

```diff
@@ -0,0 +1,70 @@
+import polars as pl
+
+from nisapi.clean.helpers import (
+    cast_types,
+    clamp_ci,
+    clean_4_level,
+    clean_geography,
+    drop_suppressed_rows,
+    enforce_columns,
+    hci_to_cis,
+    replace_overall_domain,
+    set_lowercase,
+    week_ending_to_times,
+)
+
+
+def rename_indicator_columns(df: pl.LazyFrame) -> pl.LazyFrame:
+    """
+    Make "indicator" follow the same logic as "geography" and
+    "domain", with "type" and "value" columns
+    """
+    return df.rename(
+        {
+            "geography_level": "geography_type",
+            "geography_name": "geography",
+            "demographic_level": "domain_type",
+            "demographic_name": "domain",
+            "indicator_label": "indicator_type",
+            "indicator_category_label": "indicator",
+        }
+    )
+
+
+def merge_age_groups(df: pl.LazyFrame) -> pl.LazyFrame:
+    """
+    Make the "age_group" column part of the "domain" and "domain_type" columns.
+    """
+    return (
+        df.with_columns(
+            domain_type=("age & " + pl.col("domain_type")),
+            domain=pl.concat_str(["age_group", "domain"], separator=" & "),
+        )
+        .with_columns(
+            domain_type=pl.when(pl.col("domain_type") == "age & age")
+            .then(pl.lit("age"))
+            .otherwise(pl.col("domain_type")),
+            domain=pl.when(pl.col("domain_type") == "age & age")
+            .then(pl.col("age_group"))
+            .otherwise(pl.col("domain")),
+        )
+        .drop("age_group")
+    )
```

Comment:
```suggestion
    return (
        df.with_columns(
            domain_type=pl.when(pl.col('domain_type') == pl.lit('age'))
            .then(pl.lit('age'))
            .otherwise("age & " + pl.col("domain_type")),
            domain=pl.when(pl.col('domain_type') == pl.lit('age'))
            .then(pl.col('domain'))
            .otherwise(pl.concat_str(["age_group", "domain"], separator=" & ")
        )
        .drop("age_group")
    )
```

I think you can do this all at once, and not in two steps?

If I'm right, it's because polars does all the assignments in `.with_columns()` in parallel. This is different from dplyr's `mutate()`, which does them in sequence, which has the plus that you can do a chain of operations in defining columns in one `mutate()`, but it also means you can't do an operation like this in one go

Reply:
Okay, I see now - not only are we condensing two blocks of `.with_columns()` into one, but we're also adjusting the logic therein. Works for me - fixed.